docs/reference/continuous-export.md

```diff
@@ -0,0 +1,70 @@
+---
+layout: default
+title: Continuous export
+parent: Reference
+nav_order: 10
+has_children: false
+---
+# Continuous Export
+
+You can configure a lakeFS repository can store the latest of a branch on an external
+object store using its native paths and access methods.  This allows clients and tools
+read-only access to repository objects without any lakeFS-specific configuration.
+Initially onlya AWS S3 is supported.
+
+For instance, the contents `lakefs://example@master` might be stored on
+`s3://company-bucket/example/latest`.  Clients entirely unaware of lakeFS could use that
+base URL to access latest files on `master`.  Clients aware of lakeFS can continue to use
+the lakeFS S3 endpoint to access repository files on `s3://example/master`, as well as
+other versions and uncommitted versions.
+
+## Configuration
+
+To configure continuous export for branch `master`, run
+
+```shell
+lakefs repo export --branch master --start --export-path s3://company-bucket/path/to/export/ [--export-status-path s3://company-bucket/path/to/status]
+```
+
+To stop continuous export, run
+
+```shell
+lakefs repo export --branch master --stop
+```
+
+### Alternatives
+
+#### Multiple branches
+
+If we go with per-branch export this is almost the exact same command!
+
+```shell
+lakefs branch export --branch master --start --export-path s3://company-bucket/path/to/export/
+```
+
+#### One-time export
+
+Either get rid of the `--start` or just support one-time export as a command!
+
+## Operation
+
+When continuous export is enabled for a branch its current state is exported to S3.  From
+then on, every commit or merge operation to that branch will be exported _*optional:* just
+as though a non-continuous export were run_.  The commit or merge operation on the branch
+returns as soon as export starts.
+
+The export process runs for a while.  To help track its status it creates a file
+`_STATUS_<timestamp>.json`.  This file contains multiple JSON records, each on a single
```

Comment:
we should think if we want to use commit id or timestamp as part of the name... each has its own benefits

Reply:
I also thought about it!  I think timestamp is better because it sorts filenames in meaningful order.  Because commit is so important, it appears as a field in the first record.  So we don't lose it (but admittedly it may require some work to discover the commit ID).
I think it should _not_ be part of the filename: commit ID is relevant on the lakeFS side and S3-only users should not have to deal with it if they do not wish to.

Reply:
[after talking f2f] Either way need to provide the merger / committer with details of the filename.  (Even if the filename were based on the commit ID, the merger / committer cannot predict what that ID will be, it must be returned!)

Reply:
I recommend (not sure if in addition or as substitute), adding a single file success indicator per commit. this makes it easy for orchestration tools like airflow to know when data is ready to be consumed

Reply:
I recommend (not sure if in addition or as substitute), adding a single file success indicator per commit. this makes it easy for orchestration tools like airflow to know when data is ready to be consumed

Reply:
I'm fine with any choice here as long as it makes sense.  A single file 

So I want to start with a single file `_SUCCESS` with these fields.  What can I do on failure?  It makes sense for a one-time export, but less so for a repeated export.  (That's why Hadoop can safely do it, of course).

Reply:
It could be a single success file with the same name we provided the merger / committer.
meaning we will have  _STATUS_<timestamp>.json and _SUCCESS_<timestamp>.
where <timestamp> could be commitID, timestamp or both. I don't think it matters as long as the merger / commiter will now the name of the expected file.

Reply:
about the file name -  am not sure we can change the object name by adding timestamp or commit ID.